File: src/main/java/com/abysalto/middleware/source/dummyjson/dto/DummyProduct.java
package com.abysalto.middleware.source.dummyjson.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

/**
 * Raw DummyJSON product payload. Kept package-internal to the {@code dummyjson} source so the
 * upstream shape never leaks into the rest of the application. Unknown fields are ignored so new
 * upstream attributes do not break deserialization.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record DummyProduct(
        long id,
        String title,
        String description,
        String category,
        Double price,
        Double discountPercentage,
        Double rating,
        Integer stock,
        List<String> tags,
        String brand,
        String sku,
        Double weight,
        DummyDimensions dimensions,
        String warrantyInformation,
        String shippingInformation,
        String availabilityStatus,
        String returnPolicy,
        Integer minimumOrderQuantity,
        String thumbnail,
        List<String> images,
        List<DummyReview> reviews,
        DummyMeta meta
) {
}

File: dotnet/src/Middleware.Infrastructure/Upstream/Dto/DummyProduct.cs
namespace Middleware.Infrastructure.Upstream.Dto;

/// <summary>
/// Raw DummyJSON product payload. Kept <c>internal</c> to the upstream adapter so the upstream shape
/// never leaks into the rest of the application. Unknown JSON fields are ignored by default
/// (System.Text.Json), so new upstream attributes do not break deserialization.
/// <para><see cref="Price"/> is bound directly as <see cref="decimal"/>: the JSON number is parsed
/// straight into the domain's money type without a lossy <c>double</c> hop.</para>
/// </summary>
internal sealed record DummyProduct(
    long Id,
    string? Title,
    string? Description,
    string? Category,
    decimal? Price,
    double? DiscountPercentage,
    double? Rating,
    int? Stock,
    IReadOnlyList<string>? Tags,
    string? Brand,
    string? Sku,
    double? Weight,
    DummyDimensions? Dimensions,
    string? WarrantyInformation,
    string? ShippingInformation,
    string? AvailabilityStatus,
    string? ReturnPolicy,
    int? MinimumOrderQuantity,
    string? Thumbnail,
    IReadOnlyList<string>? Images,
    IReadOnlyList<DummyReview>? Reviews,
    DummyMeta? Meta);
